tests/test_blocks/test_step_impl/test_ek1_projmatfree_d_nu.py

```diff
@@ -0,0 +1,103 @@
+"""Tests for traditional EK1 implementations."""
+
+from functools import partial
+
+import jax
+import jax.numpy as jnp
+import pytest
+
+from tornadox.blocks.sde import ibm
+from tornadox.blocks.step_impl.first_order import ek1_projmatfree_d_nu
+
+
+@partial(jax.jit, static_argnames=("n", "d"))
+def _setup(*, n, d):
+    """Setup for derivative-dimension ordering."""
+
+    # Preconditioners
+    p, p_inv = ibm.preconditioner_diagonal(dt=0.1, num_derivatives=n)
+    p, p_inv = jnp.tile(p, reps=d), jnp.tile(p_inv, reps=d)
+
+    # System matrices
+    eye_d = jnp.eye(d)
+    a, q_sqrtm = ibm.system_matrices_1d(num_derivatives=n)
+    a, q_sqrtm = jnp.kron(eye_d, a), jnp.kron(eye_d, a)
```

Comment:
Seeing this and the above: Shouldn't we maybe add a dimension keyword to the ibm stuff? So far I don't have the impression it would make the code too complex or hard to use/maintain :sweat_smile:

Reply:
Not sure...
* It would move the decision of dimension-derivative ordering to the `ibm` module, not to each solver. That is, unless we do both, in which case there are suddenly 2 more functions to maintain, which are only rarely used:
* Only very few solvers actually go full-Kronecker. So I am not sure how many lines of code this actually saves

What do you think?

Reply:
- do we ever want different dimension-derivative orderings? If no I would be fine with deciding on one and having ibm handle this. If we need both, I agree that that might create not so nice code and maybe letting the "user" handle it is easier.
- I would argue that building the full matrix is the most standard thing to do. But I see the point with "not saving lines of code" since it's only really the EK1.

Reply:
Let's leave it as is, because for now the package doesn't need it, but once we realise it is needed repeatedly, we factor this out appropriately?